gobin/exe.go

```diff
@@ -174,3 +169,42 @@ func toPackages(ctx context.Context, out *[]*claircore.Package, p string, r io.R
 		Msg("analyzed exe")
 	return nil
 }
+
+var versionRegex = regexp.MustCompile(`^v?([0-9]+)(\.[0-9]+)?(\.[0-9]+)?(-([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?$`)
+var ErrInvalidSemVer = errors.New("invalid semantic version")
+
+// ParseVersion will return a claircore.Version of type semver given
+// a valid semantic version. If the string is not a valid semver it
+// will return an ErrInvalidSemVer.
+func ParseVersion(ver string) (c claircore.Version, err error) {
```

Comment:
In the case of a go pseudo version, should it discard the date? I think I could go either way.

Reply:
`ParseVersion()` behaves the same as `semver.NewVersion()` + `version.FromSemver()` in the current code so I believe anything that is not in the Major, Minor, Patch triplet is discarded. If I'm not mistaken we're talking about versions like this: 
```
v0.0.0-20170426191122-ca1404ee6e83
```
In which case I'm not sure how we fit that into the current `claircore.Version` so it's possibly moot. In future world, maybe it makes sense to cut off the hash and save the date (presumably there is some general acceptance that `v0.0.0-20170426191122-aaaaaaaaaaaa > v0.0.0-20170426191121-zzzzzzzzzzzz`)

Reply:
Just for any observers, the problem versions are currently versions like 
```
v1.2022081210521689159162.2023071210561689159372
```
For "valid" semvers like:
```
v0.0.0-20170426191122-ca1404ee6e83
``` 
they are currently parsed as just `v0.0.0` (or `0` to be more precise)